samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java

```diff
@@ -1264,17 +1264,34 @@ private double getTotalAutoScore(Set itemGradingSet){
     return totalAutoScore;
   }
 
-  private void notifyGradebookByScoringType(AssessmentGradingData data, PublishedAssessmentIfc pub){
+  public void notifyGradebookByScoringType(AssessmentGradingData data, PublishedAssessmentIfc pub){
+    if (pub == null || pub.getEvaluationModel() == null) {
+      // should not come to here
+      log.warn("publishedAssessment is null or publishedAssessment.getEvaluationModel() is null");
+      return;
+    }
     Integer scoringType = pub.getEvaluationModel().getScoringType();
     if (updateGradebook(data, pub)){
       AssessmentGradingData d = data; // data is the last submission
       // need to decide what to tell gradebook
-      if ((scoringType).equals(EvaluationModelIfc.HIGHEST_SCORE))
+      if ((scoringType).equals(EvaluationModelIfc.HIGHEST_SCORE)) {
         d = getHighestSubmittedAssessmentGrading(pub.getPublishedAssessmentId().toString(), data.getAgentId());
+      }
+      // Send the average score if average was selected for multiple submissions
+      else if (scoringType.equals(EvaluationModelIfc.AVERAGE_SCORE)) {
+        // status = 5: there is no submission but grader update something in the score page
+        if(data.getStatus() ==5) {
```

Comment:
Can you use the constant AssessmentGradingData.NO_SUBMISSION here?

Reply:
Yeah, I can update that, I just moved this code block without updating these constants.